### PR TITLE
Fix issue #581

### PR DESCRIPTION
**Pull request by anawas**
* Import of OSKAR telescope files with function `read_OSKAR_tm_file` now works.
* I refactored a lot of tests in `test/test_telescope.py`. They used `plot_telescope()` in order test if the Telescope instance was created successfully. This took a lot fo time, especially when only this test was run. The test functions now test other telescope specific values.

### CODE REVIEW

**Comment by Lukas113** [comment]
I'll wait for the review until #601 is merged since #601 is also in this PR.

**Comment by mpluess** [comment]
@anawas can you merge main into this branch and solve the conflicts? i will review it after that's done.

**Review by mpluess — CHANGES_REQUESTED** [review]
Looks good. I like that you reduced the test runtime by removing the plot_telescope calls, but i think it would make sense to add two dedicated tests calling plot_telescope to make sure it works, one with OSKAR, one with RASCIL backend since they have different code paths in plot_telescope. The code path with an invalid backend is already covered in the tests.

File: karabo/simulation/telescope.py [hunk 1]
```diff
@@ -402,7 +402,14 @@ def plot_telescope(self, file: Optional[str] = None) -> None:
         if self.backend is SimulatorBackend.OSKAR:
             self.plot_telescope_OSKAR(file)
         elif self.backend is SimulatorBackend.RASCIL:
-            plot_configuration(self.get_backend_specific_information())
+            if file is not None:
```
Comment (mpluess) [inline karabo/simulation/telescope.py:405]: if-else seems unnecessary, default value of plot_configuration's plot_file is None, so you should be able to just pass plot_file=file in both cases

Comment (anawas) [reply karabo/simulation/telescope.py]: Solved in commit 75fe915

**Comment by mpluess** [comment]
Changes look good so far, tests are failing though

**Comment by anawas** [comment]
That's what I talked about on Slack. I did not touch the file the pre-commit hook is complaining about. I ran the tests locally and they passed. This pre-commit hook is bit of hard to deal with.

**Comment by mpluess** [comment]
> That's what I talked about on Slack. I did not touch the file the pre-commit hook is complaining about. I ran the tests locally and they passed. This pre-commit hook is bit of hard to deal with. Screenshot 2024-08-02 115004

I seems hard to have 100 % the same environment as the CI tests on GitHub. I think all of us devs have had to fix things in reaction to tests failing on GitHub before. Had to remove the same "type: ignore" comment in another branch this morning. One difference is probably that on GitHub, a new environment is created on every action, meaning our dev envs are probably a bit behind on same package versions. Updating the environment more often could open.

**Review by mpluess — APPROVED** [review]
LGTM

**Pull request merged**